src/util/matcher/stitcher.js

```diff
@@ -0,0 +1,104 @@
+function stitcher(processedPoints, overlay, map) { // jshint ignore:line
+  var center, corners;
+  var sectors = {s00: [], s01: [], s10: [], s11: [], population: []};
+  for (var i in processedPoints.points) {
+    sectors.s00[i] = [];
+    sectors.s01[i] = [];
+    sectors.s10[i] = [];
+    sectors.s11[i] = [];
+    center = processedPoints.images[i].getCenter();
+    corners = processedPoints.images[i].getCorners();
+    for (var j in processedPoints.points[i]) {
+      if (
+        processedPoints.points[i][j].lng > corners[1].lng &&
+        processedPoints.points[i][j].lng < center.lng
+      ) {
+        if (
+          processedPoints.points[i][j].lat > corners[3].lat &&
+          processedPoints.points[i][j].lat < center.lat
+        ) {
+          sectors.s10[i].push(processedPoints.points[i][j]);
+        } else {
+          sectors.s00[i].push(processedPoints.points[i][j]);
+        }
+      } else {
+        if (
+          processedPoints.points[i][j].lat > corners[2].lat &&
+          processedPoints.points[i][j].lat < center.lat
+        ) {
+          sectors.s11[i].push(processedPoints.points[i][j]);
+        } else {
+          sectors.s01[i].push(processedPoints.points[i][j]);
+        }
+      }
+    }
+  }
+  for (i=0; i<processedPoints.points.length; i++) {
+    sectors.population[i] = [];
+    sectors.population[i].push([
+      sectors.s00[i].length,
+      sectors.s01[i].length,
+      sectors.s10[i].length,
+      sectors.s11[i].length
+    ]);
+  }
+  var max = 0,
+    min = 10,
+    max_ = 0,
+    max_idx;
+  for (i in sectors.population[0][0]) {
+    if (sectors.population[0][0][i] > max) {
+      max = sectors.population[0][0][i];
+      max_idx = i;
+    }
+    if (sectors.population[0][0][i] < min) {
+      min = sectors.population[0][0][i];
+    }
+  }
+  if (max !== min) {
+    var coordinates = sectors[Object.keys(sectors)[max_idx]][0];
+    for (var u in processedPoints.points[0]) {
+      for (var v in coordinates) {
+        if (processedPoints.points[0][u] === coordinates[v]) {
+          if (processedPoints.confidences[0][u] > max_) {
+            max_ = processedPoints.confidences[0][u];
+          }
+        }
+      }
+    }
+    var best_point =
```

Comment:
Could some of this be refactored into a subfunction called something like `filterBestPoints()` or something? That wouldn't have to mix at all with HTML as this file does?

Reply:
Hmm. Do you want me to expose an API here, or should this change be incorporated implicitly?

Reply:
Sure, should I extract this fn into a different file too?

Reply:
Well if the ability to filter points gets moved back into the core with `projector` and doesn't mix with the DOM, does that make sense? It could be an exposed API so that the point selection metric could be overridden...